### ISSUE TITLE
TypeScript: comment is moved between modifier and function

### ISSUE THREAD

**Issue by frankwallis** [prettier/prettier#2727]
This:

```
    @decorateMyEvent('some params which are long enough to push everything over')
    // tslint:disable-next-line
    private _onSelectedRowsChangedHandler({ selectedRows }: SelectedRowsChangedEvent<TRowData>): void {
```
becomes
```
    @decorateMyEvent('some params which are long enough to push everything over')
    private // tslint:disable-next-line
    _onSelectedRowsChangedHandler({ selectedRows }: SelectedRowsChangedEvent<TRowData>): void {
```
Which messes everything up as 'private' is now seen as a class member and the tslint-ignore stops working etc.

**Comment by asmockler** [comment]
I think this was closed with #2906 - can anyone confirm?

**Comment by azz** [comment]
Confirmed! Thanks.

**Prettier 1.7.4**
[Playground link](https://prettier.io/playground/#%7B%22content%22%3A%22class%20X%20%7B%5Cn%20%20%20%20%20%20%40decorateMyEvent('some%20params%20which%20are%20long%20enough%20to%20push%20everything%20over')%5Cn%20%20%20%20%2F%2F%20tslint%3Adisable-next-line%5Cn%20%20%20%20private%20_onSelectedRowsChangedHandler(%7B%20selectedRows%20%7D%3A%20SelectedRowsChangedEvent%3CTRowData%3E)%3A%20void%20%7B%5Cn%20%20%20%20%7D%5Cn%7D%22%2C%22options%22%3A%7B%22ast%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22output2%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22printWidth%22%3A80%2C%22semi%22%3Atrue%2C%22singleQuote%22%3Afalse%2C%22tabWidth%22%3A2%2C%22trailingComma%22%3A%22none%22%2C%22useTabs%22%3Afalse%7D%7D)
```sh
--parser typescript
```

**Input:**
```tsx
class X {
      @decorateMyEvent('some params which are long enough to push everything over')
    // tslint:disable-next-line
    private _onSelectedRowsChangedHandler({ selectedRows }: SelectedRowsChangedEvent<TRowData>): void {
    }
}
```

**Output:**
```tsx
class X {
  @decorateMyEvent("some params which are long enough to push everything over")
  // tslint:disable-next-line
  private _onSelectedRowsChangedHandler({
    selectedRows
  }: SelectedRowsChangedEvent<TRowData>): void {}
}

```
